gnomad_qc/v3/create_release/prepare_vcf_data_release.py

```diff
@@ -0,0 +1,1019 @@
+                )
+            prepared_vcf_ht = hl.read_table(parameter_dict["temp_ht_path"])
+
+        if args.prepare_vcf_header_dict:
+            logger.info("Making histogram bin edges...")
+            bin_edges = make_hist_bin_edges_expr(
+                parameter_dict["ht"],
+                prefix="gnomad" if hgdp_1kg else "",
+                include_age_hists=parameter_dict["include_age_hists"],
+            )
+
+            header_dict = prepare_vcf_header_dict(
+                prepared_vcf_ht,
+                bin_edges=bin_edges,
+                age_hist_data=parameter_dict["age_hist_data"],
+                subset_list=parameter_dict["subsets"],
+                pops=parameter_dict["pops"],
+            )
+            if not hgdp_1kg:
+                header_dict.pop("format")
+
+            logger.info("Saving header dict to pickle...")
+            with hl.hadoop_open(
+                release_header_path(subset="hgdp_1kg" if hgdp_1kg else None), "wb",
+            ) as p:
+                pickle.dump(header_dict, p, protocol=pickle.HIGHEST_PROTOCOL)
+
+        if args.sanity_check:
+            logger.info("Beginning sanity checks on the prepared VCF HT...")
+            sanity_check_release_ht(
+                prepared_vcf_ht,
+                subsets=["gnomad"] if hgdp_1kg else SUBSETS,
+                subset_before_metric=True if hgdp_1kg else False,
+                subset_pops=parameter_dict["subset_pops"],
+                missingness_threshold=0.5,
+                verbose=args.verbose,
+            )
+
+        if args.export_vcf:
+            logger.info("Loading VCF header dict...")
+            with hl.hadoop_open(
+                release_header_path(subset="hgdp_1kg" if hgdp_1kg else None), "rb",
+            ) as f:
+                header_dict = pickle.load(f)
+
+            logger.info("Cleaning up the VCF HT for final export...")
+            prepared_vcf_ht, new_row_annots = cleanup_ht_for_vcf_export(
+                prepared_vcf_ht,
+                drop_freqs=hl.eval(prepared_vcf_ht.freq_entries_to_remove),
+                drop_hists=DROP_HISTS,
+            )
+
+            logger.info("Running check on VCF fields and info dict...")
+            if not vcf_field_check(prepared_vcf_ht, header_dict, new_row_annots):
+                logger.error("Did not pass VCF field check")
+
+            if hgdp_1kg:
+                logger.info(
+                    "Loading the HGDP + TGP subset MT and annotating with the prepared VCF HT for VCF export..."
+                )
+                t = release_subset(subset="hgdp_1kg", dense=True).mt().select_rows()
+                t = t.annotate_rows(**prepared_vcf_ht[t.row_key])
+            else:
+                t = prepared_vcf_ht
+
+            if args.test:
+                t = filter_to_test(t)
+
+            if chromosome:
+                t = hl.filter_intervals(t, [hl.parse_locus_interval(chromosome)])
```

Comment:
do you need to specify that the reference is `export_reference` here?

Reply:
This is done before `rekey_new_reference(t, export_reference` though so I think this is OK